intelmq/bots/outputs/rpz_file/output.py

```diff
@@ -0,0 +1,154 @@
+from pathlib import Path
+
+from intelmq.lib.bot import OutputBot
+
+RPZ_INDICATOR_MAP = {
+    "source.fqdn": "Intel::DOMAIN"
+}
+
+now = datetime.now()
+
+
+class RpzFileOutputBot(OutputBot):
+    _file = None
+    format_filename: bool = False
+    hierarchical_output: bool = False
+    keep_raw_field: bool = False
+    message_jsondict_as_string: bool = False
+    message_with_type: bool = False
+    single_key: bool = False
+    is_multithreadable = False
+
+    encoding_errors_mode = 'strict'
+    file: str = "/opt/intelmq/var/lib/bots/file-output/rpz"
+
+    cname: str = ""
+    organization_name: str = ''
+    rpz_domain: str = ''
+    hostmaster_rpz_domain: str = ''
+    rpz_email: str = ''
+    ttl: str = '3600'
+    generate_time: str = now.strftime("%Y-%m-%d %H:%M:%S")
+    ncachttl: str = '60'
+    serial: str = now.strftime("%y%m%d%H%M")
+    refresh: str = '60'
+    retry: str = '60'
+    expire: str = '432000'
+    test_domain: str = ''
+
+    rpz_header = ""
+
+    def init(self):
+        self._file = None
+
+        self.set_rpz_header()
+
+        self.logger.debug("Opening %r file.", self.file)
+        self.format_filename = getattr(self, 'format_filename', False)
+        self.errors = getattr(self, 'encoding_errors_mode', 'strict')
+        if not self.format_filename:
+            self.open_file(self.file)
+        self.logger.info("File %r is open.", self.file)
+
+    def set_rpz_header(self):
+        self.rpz_header = "$TTL " + str(self.ttl) + "\n@ SOA " + self.rpz_domain + ". " + self.hostmaster_rpz_domain + ". " + str(self.serial) + " " + str(self.refresh) + " " + str(self.retry) + " " + str(self.expire) + " " + str(self.ncachttl) + "\n NS localhost.\n;\n; " + self.organization_name + " Response Policy Zones (RPZ)\n; Last updated: " + str(self.generate_time) + " (UTC)\n;\n; Terms Of Use: https://" + self.rpz_domain + "\n; For questions please contact " + self.rpz_email + "\n;\n"
+        if self.test_domain:
+            self.rpz_header = self.rpz_header + self.test_domain + " CNAME " + self.cname + ".\n"  # for test
+
+    def open_file(self, filename: str = None):
+        if self._file is not None:
+            self._file.close()
+
+        try:
+            self._file = open(filename, mode='a+t', encoding='utf-8', errors=self.errors)
+            self.add_rpz_header()
+        except FileNotFoundError:  # directory does not exist
+            path = Path(os.path.dirname(filename))
+            try:
+                path.mkdir(mode=0o755, parents=True, exist_ok=True)
+            except IOError:
+                self.logger.exception('Directory %r could not be created.', path)
+                self.stop()
+            else:
+                self._file = open(filename, mode='a+t', encoding='utf-8', errors=self.errors)
+                self.add_rpz_header()
+
+    def add_rpz_header(self):
+        self._file.seek(0)
+        if self._file.read() != self.rpz_header and not os.stat(self._file.name).st_size:
```

Comment:
I don't understand the logic of this line. First, you check that the *complete* file is unequal to the previously generated RPZ header, then you stat by the filename to ensure it is empty. Wouldn't a size-check suffice then? (which can be done with the `Path` object as well, btw)

Reply:
Just check if the file is empty.

Reply:
Does this work as well?
```suggestion
        if not (len(self._file.read())):
```